resnet-burn/src/model/resnet.rs

```diff
@@ -171,6 +258,172 @@ impl<B: Backend> ResNet<B, Bottleneck<B>> {
     ///
     /// A ResNet-152 module.
     pub fn resnet152(num_classes: usize, device: &Device<B>) -> Self {
-        Self::new([3, 8, 36, 3], num_classes, 4, device)
+        ResNetConfig::<B, Bottleneck<B>>::new(RESNET152_BLOCKS, num_classes, 4).init(device)
+    }
+
+    /// ResNet-152 from [`Deep Residual Learning for Image Recognition`](https://arxiv.org/abs/1512.03385)
+    /// with pre-trained weights.
+    ///
+    /// # Arguments
+    ///
+    /// * `weights`: Pre-trained weights to load.
+    /// * `device` - Device to create the module on.
+    ///
+    /// # Returns
+    ///
+    /// A ResNet-152 module with pre-trained weights.
+    #[cfg(feature = "pretrained")]
+    pub fn resnet152_pretrained(
+        weights: weights::ResNet50,
+        device: &Device<B>,
+    ) -> Result<Self, RecorderError> {
+        let weights = weights.weights();
+        let record = Self::load_weights_record(&weights, device)?;
+        let model = ResNetConfig::<B, Bottleneck<B>>::new(RESNET152_BLOCKS, weights.num_classes, 1)
+            .init_with(record);
+
+        Ok(model)
+    }
+}
+
+/// [ResNet](ResNet) configuration.
+struct ResNetConfig<B, M> {
+    conv1: Conv2dConfig,
+    bn1: BatchNormConfig,
+    maxpool: MaxPool2dConfig,
+    layer1: LayerBlockConfig<M>,
+    layer2: LayerBlockConfig<M>,
+    layer3: LayerBlockConfig<M>,
+    layer4: LayerBlockConfig<M>,
+    avgpool: AdaptiveAvgPool2dConfig,
+    fc: LinearConfig,
+    _backend: PhantomData<B>,
+}
+
+impl<B: Backend, M> ResNetConfig<B, M> {
+    /// Create a new instance of the ResNet [config](ResNetConfig).
+    pub fn new(blocks: [usize; 4], num_classes: usize, expansion: usize) -> Self {
+        // `new()` is private but still check just in case...
```

Comment:
Is this still true? new seems pub to me

Reply:
And it totally is 🤣 it was private before the refactor, will fix this discrepancy lol